compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -429,19 +455,34 @@ static SmallVector<int64_t> getDefaultDistributedLoopTileSizes(
   return distributedTileSizes;
 }
 
+/// Returns the nearest power of two of `size` if `predicate` is true.
+/// Otherwise, returns `size`.
+static int64_t roundUpToPow2(int64_t size, bool predicate) {
+  if (!predicate) {
+    return size;
+  }
+  assert(size > 0 && "Negative size");
+  return 1 << llvm::Log2_64_Ceil(size);
+}
+
 /// Adjusts the workload per workgroup to be a multiple of vector size to ensure
 /// that the op vectorizes.
 static int64_t getMaxTileSize(int64_t lb, int64_t ub, int64_t maxSize,
                               int64_t vectorSize,
-                              bool allowIncompleteTile = false) {
+                              bool allowIncompleteTile = false,
+                              bool enforcePowerOfTwo = false) {
   if (ub == ShapedType::kDynamic || lb == ShapedType::kDynamic) {
-    return maxSize;
+    return roundUpToPow2(maxSize, enforcePowerOfTwo);
   }
   int64_t numIters = ub - lb;
   if (numIters <= maxSize && numIters < vectorSize) {
-    return numIters;
+    return roundUpToPow2(numIters, enforcePowerOfTwo);
   }
 
+  // Return the largest suitable power of two if power of two is enforced.
+  if (enforcePowerOfTwo)
+    return roundUpToPow2(std::min(maxSize, numIters), enforcePowerOfTwo);
+
```

Comment:
style nit: warp with braces

```suggestion
  if (enforcePowerOfTwo) {
    return roundUpToPow2(std::min(maxSize, numIters), enforcePowerOfTwo);
  }

```

Reply:
It would be great if we could reconsider these mismatches in style with MLIR/LLVM to make the codebase more consistent with open source. Perhaps something to discuss in the weekly meeting.

Reply:
Or maybe both should be allowed in both places 😆